examples/js/WaterShader.js

```diff
@@ -140,16 +140,20 @@ THREE.Water = function ( renderer, camera, scene, options ) {
 			'}',
 
 			THREE.ShaderChunk[ 'common' ],
+			THREE.ShaderChunk[ 'packing' ],
 			THREE.ShaderChunk[ 'fog_pars_fragment' ],
+			THREE.ShaderChunk[ 'lights_pars' ],
+			THREE.ShaderChunk[ 'shadowmap_pars_fragment' ],
+			THREE.ShaderChunk[ 'shadowmask_pars_fragment' ],
 
 			'void main() {',
-			'	vec4 noise = getNoise( worldPosition.xz );',
+			'	vec4 noise = getNoise( worldPosition.xz * size );',
 			'	vec3 surfaceNormal = normalize( noise.xzy * vec3( 1.5, 1.0, 1.5 ) );',
 
 			'	vec3 diffuseLight = vec3(0.0);',
 			'	vec3 specularLight = vec3(0.0);',
 
-			'	vec3 worldToEye = eye-worldPosition;',
+			'	vec3 worldToEye = eye-worldPosition.xyz;',
 			'	vec3 eyeDirection = normalize( worldToEye );',
 			'	sunLight( surfaceNormal, eyeDirection, 100.0, 2.0, 0.5, diffuseLight, specularLight );',
```

Comment:
Does it still make sense to support `sunDirection` and `sunColor` when the material reacts on light source now?

Reply:
I think it does, the bsdfs are still not used except the one of the shader, that gives his look. The new chunks are only here to support the shadowmap. However your question made me check with different lights, and the shader is currently broken if we use other types of lights than the directional lights. Working on it.